lp2p/relaynode.go

```diff
@@ -134,40 +134,44 @@ func ParseMultiaddrSlice(peers []string) ([]ma.Multiaddr, error) {
 func (g *GossipRelayNode) background(c client.Client) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	results := c.Watch(ctx)
 	for {
-		select {
-		case res, ok := <-results:
-			if !ok {
+		results := c.Watch(ctx)
```

Comment:
maybe add time.Sleep(time.Second), to prevent thrashing if the implementation of c.Watch() happens to always return a channel (even when it is about to close immediately). I've seen this pattern in a few places.